### PR TITLE
[bitnami/mongodb-sharded] numactl flag for numa hardware

### LINKED ISSUE #6660: [bitnami/mongodb-sharded] numactl issue in all instances
**Which chart**:
bitnami/mongodb-sharded:4.5.0

**Is your feature request related to a problem? Please describe.**
logging into a mogodb through a POD, client complains NUMA settings are missing,

**Describe the solution you'd like**
Is it possible add an external variable in values.yaml that can help to launch instances with numactl set as complained?

**Additional context**
The following lines with the issue from client:
```
The server generated these startup warnings when booting: 
        2021-06-14T14:41:46.196+00:00: You are running on a NUMA machine. We suggest launching mongod like this to avoid performance problems: numactl --interleave=all mongod [other options]
        2021-06-14T14:41:46.196+00:00: /sys/kernel/mm/transparent_hugepage/enabled is 'always'. We suggest setting it to 'never'
```

Thx
F43RY

### PR DESCRIPTION
**Pull request by F43RY**
**Description of the change**
The change add the possibility to prepend "numactl --interleave=all" command to mongod instances added in the following PR:
[PR17](https://github.com/bitnami/bitnami-docker-mongodb-sharded/pull/17)

**Benefits**

- In NUMA hardware this capability increases performances.

- Removes initial Warning in console when numactl is not executed in numa hardware

Remove

**Possible drawbacks**

**Applicable issues**

  - fixes #6660 

**Additional information**
MongoDB official reference on numactl [MongoDB Production notes](https://docs.mongodb.com/manual/administration/production-notes/#mongodb-and-numa-hardware)

**Checklist** 

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/).
- [X] Variables are documented in the README.md
- [X] Title of the PR starts with chart name (e.g. [bitnami/<name_of_the_chart>])

### CODE REVIEW

**Comment by migruiz4** [comment]
Hi @F43RY,

I'm sorry but the image `bitnami/mongodb-sharded` does not have a feature that is enabled by the variable `MONGODB_ENABLE_NUMACTL`. 

Does this PR have a complementary PR adding that feature for the [`mongodb-sharded` container image](https://github.com/bitnami/bitnami-docker-mongodb-sharded/)? If so, please add a link in the description.

**Comment by F43RY** [comment]
Hi @migruiz4,
just added [PR](https://github.com/bitnami/bitnami-docker-mongodb-sharded/pull/17) in description above

File: bitnami/mongodb-sharded/Chart.yaml [hunk 1]
```diff
@@ -26,4 +26,4 @@ name: mongodb-sharded
 sources:
   - https://github.com/bitnami/bitnami-docker-mongodb-sharded
   - https://mongodb.org
-version: 3.8.3
+version: 3.8.4
```
Comment (migruiz4) [inline bitnami/mongodb-sharded/Chart.yaml:29]: Please bump a minor version (3.9.0)

File: bitnami/mongodb-sharded/README.md [hunk 2]
```diff
@@ -87,6 +87,7 @@ The command removes all the Kubernetes components associated with the chart and
 | `usePasswordFile`                     | Mount credentials as files instead of using environment variables                                                                                         | `false`                   |
 | `shards`                              | Number of shards to be created                                                                                                                            | `2`                       |
 | `common.useHostnames`                 | Enable DNS hostnames in the replica set config                                                                                                            | `true`                    |
+| `common.mongodbEnableNumactl`         | Enable "numactl --interleave=all" before mongd instances                                                                                                  | `false`                    |
```
Comment (migruiz4) [inline bitnami/mongodb-sharded/README.md:90]: Small typo in `mongd`.
Additionally, please use the same description as in the `values.yaml`:
```
Enable launch MongoDB instance prefixed with "numactl --interleave=all"
```

File: bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml [hunk 3]
```diff
@@ -97,6 +97,12 @@ spec:
             - containerPort: {{ $.Values.service.port }}
               name: mongodb
           env:
+            - name: MONGODB_ENABLE_NUMACTL
+              {{- if $.Values.common.mongodbEnableNumactl }}
+              value: "yes"
+              {{- else }}
+              value: "no"
+              {{- end }}
```
Comment (migruiz4) [inline bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml:105]: Please use [ternary](https://helm.sh/docs/chart_template_guide/function_list/#ternary) instead:

```suggestion
            - name: MONGODB_ENABLE_NUMACTL
              value: {{ ternary "yes" "no" $.Values.common.mongodbEnableNumactl }}
```

Comment (migruiz4) [reply bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml:105]: Also, I'm not familiar with the Numa features, but shouldn't this also be added to all the Statefulsets? (mongos, config, arbiter and data)

I see config and data statefulsets, but I'm missing `mongos` and `arbiter`.

From [MongoDB documentation](https://docs.mongodb.com/manual/administration/production-notes/#configuring-numa-on-linux), sounds like every mongod and mongos instances require this setting, right?
> Ensure that mongod and mongos are started by numactl.

**Review by migruiz4 — CHANGES_REQUESTED** [review]
Thanks for referencing the mongodb-sharded PR @F43RY !

I left some comments below.

File: bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml (hunk 3 above)
Comment (F43RY) [reply bitnami/mongodb-sharded/templates/shard/shard-data-statefulset.yaml]: You're right.
Added to arbiter and mongos and changed to ternary in config statefulset too.

**Review by miguelaeh — APPROVED** [review]
LGTM!
We need to wait until the image with the proper changes is released

**Pull request merged**